File: Medium/2483. Minimum Penalty for a Shop/2483. Minimum Penalty for a Shop.java
class Solution {
    public int bestClosingTime(String customers) {
        int maxScore = 0;
        int bestHour = -1;
        int score = 0;

        for (int hour = 0; hour < customers.length(); hour++) {
            char c = customers.charAt(hour);
            score += (c == 'Y') ? 1 : -1;
            if (score > maxScore) {
                maxScore = score;
                bestHour = hour;
            }
        }

        return bestHour + 1;
    }
}

File: Medium/2483. Minimum Penalty for a Shop/2483. Minimum Penalty for a Shop.js
/**
 * @param {string} customers
 * @return {number}
 */
var bestClosingTime = function (customers) {
    let maxScore = 0;
    let bestHour = -1;
    let score = 0;

    for (let hour = 0; hour < customers.length; hour++) {
        const c = customers.charAt(hour);
        score += c === 'Y' ? 1 : -1;
        if (score > maxScore) {
            maxScore = score;
            bestHour = hour;
        }
    }

    return bestHour + 1;
};
